src/Interpreters/inplaceBlockConversions.cpp

```diff
@@ -423,10 +425,11 @@ void fillMissingColumns(
         const auto * array_type = typeid_cast<const DataTypeArray *>(requested_column->type.get());
         if (array_type && !offsets_columns.empty())
         {
-            num_dimensions = array_type->getNumberOfDimensions();
+            num_dimensions = getNumberOfDimensions(*array_type);
             current_offsets.resize(num_dimensions);
 
-            auto serialization = IDataType::getSerialization(*requested_column);
+            auto serialization_info = requested_column->type->getSerializationInfo(*res_columns[i]);
```

Comment:
Tests showed that a `type` might be nullptr.

Reply:
It's actually `res_columns[i]` might be `nullptr`, I added a check for that